packages/zowe-explorer-api/src/profiles/ProfilesConfig.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the *
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at *
+ * https://www.eclipse.org/legal/epl-v20.html                                      *
+ *                                                                                 *
+ * SPDX-License-Identifier: EPL-2.0                                                *
+ *                                                                                 *
+ * Copyright Contributors to the Zowe Project.                                     *
+ *                                                                                 *
+ */
+
+import { IProfileLoaded, Logger, IProfAttrs, IProfile, ProfileInfo } from "@zowe/imperative";
+
+export interface IProfileValidationConfig {
+    status: string;
+    name: string;
+}
+
+export interface IValidationSettingConfig {
+    name: string;
+    setting: boolean;
+}
+export class ProfilesConfig {
```

Comment:
Yes, you are right that this mirrors the `ProfilesCache` implementation. This is because `ProfilesCache` will eventually be deprecated together with the old-school profiles and this new `ProfilesConfig` class will be the new "cache".

Reply:
Okay, so it makes even more sense to implement `ProfilesConfig` as a new implementation of `Profiles` interface now, with this information, at least for me.

In that case, we will not be required to change any consuming parts of ZE and extenders - but just provide a brand new implementation. It will also simplify a transition period and make an API better: in case we find something, which does not belong to the `Profiles` interface - we can improve it.